### PR TITLE
Resolve #1594, #1603

### LINKED ISSUE #1591: Only do e2e tests for pr

### LINKED ISSUE #1594: [Test] Reduce number of infura calls

### PR DESCRIPTION
**Pull request by timothywangdev**

### CODE REVIEW

File: src/tests/e2e/utils/requestInterceptor.js [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+}
+
+const LEDGER_API_TXS = 'https://api.ledgerwallet.com/blockchain/v2/btc_testnet/addresses'
+const INFURA_API = 'https://rinkeby.infura.io/v3/'
+const CHAINSFR_API = process.env.REACT_APP_CHAINSFER_API_ENDPOINT
+const TEST_URL = process.env.E2E_TEST_URL
+
+class RequestInterceptor {
+  intercept = false
+  byPassMatchObj = null
```
Comment (InVincible2016) [inline src/tests/e2e/utils/requestInterceptor.js:68]: Add flow type definition

File: src/tests/e2e/utils/requestInterceptor.js [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+  stats = {}
```
Comment (InVincible2016) [inline src/tests/e2e/utils/requestInterceptor.js:69]: Add flow type definition

File: src/tests/e2e/utils/requestInterceptor.js [hunk 3]
```diff
@@ -0,0 +1,375 @@
...
+  cache = {}
```
Comment (InVincible2016) [inline src/tests/e2e/utils/requestInterceptor.js:70]: Add flow type definition

File: src/tests/e2e/mocks/cryptoAccounts.js [hunk 4]
```diff
@@ -104,7 +104,7 @@ const CRYPTO_ACCOUNTS = [
   // ledger
   {
     addedAt: 1583802157,
-    address: '0xd3ced3b16c8977ed0e345d162d982b899e978588',
+    address: '0x6F6a68eC176AAfFE76e6771F2Ea79607f6B79801',
```
Comment (InVincible2016) [inline src/tests/e2e/mocks/cryptoAccounts.js:107]: What's the reason of making this change.

File: src/tests/e2e/jest-environment-puppeteer/custom.setup.js [hunk 5]
```diff
@@ -34,3 +34,6 @@ page
   .on('requestfailed', request =>
     console.log(chalk.magenta(`${request.failure().errorText} ${request.url()}`))
   )
+
+const RequestInterceptor = require('../utils/requestInterceptor').default
+global.requestInterceptor = new RequestInterceptor()
```
Comment (InVincible2016) [inline src/tests/e2e/jest-environment-puppeteer/custom.setup.js:39]: Add a new line at the end

File: src/tests/e2e/emailTransferFormEntryPoint.test.e2e.js [hunk 6]
```diff
@@ -46,12 +47,8 @@ describe('Email transfer form entry point tests', () => {
   it(
     'Recipients page entry',
     async () => {
-      await Promise.all([
-        page.waitForNavigation({
-          waitUntil: 'networkidle0'
-        }),
-        page.goto(`${process.env.E2E_TEST_URL}/contacts`)
-      ])
+      //await requestInterceptor.setRequestInterception(true)
```
Comment (InVincible2016) [inline src/tests/e2e/emailTransferFormEntryPoint.test.e2e.js:50]: remove if not used

File: src/tests/e2e/directTransferForm.test.e2e.js [hunk 7]
```diff
@@ -7,7 +7,15 @@ const timeout = 180000
 
 describe('Direct transfer form tests', () => {
   beforeAll(async () => {
+    /* browser.on('targetcreated', async (target) => {
```
Comment (InVincible2016) [inline src/tests/e2e/directTransferForm.test.e2e.js:10]: Remove if not used

File: src/tests/e2e/utils/requestInterceptor.js [hunk 8]
```diff
@@ -0,0 +1,375 @@
...
+            requestId
+          })
+        } catch (e) {
+          log.warn(`RequestInterceptor encounters error for request ${url}`, e)
+        }
+      }
+    )
+  }
+  
+  byPass (matchObj) {
```
Comment (InVincible2016) [inline src/tests/e2e/utils/requestInterceptor.js:348]: Add comment to this function

File: src/tests/e2e/utils/requestInterceptor.js [hunk 9]
```diff
@@ -0,0 +1,375 @@
...
+  byPassMatchObj = null
+  stats = {}
+  cache = {}
+
+  async setRequestInterception (val) {
+    this.intercept = val
+    this.startIntercept()
+  }
+
+  incrementStat (key, val) {
```
Comment (InVincible2016) [inline src/tests/e2e/utils/requestInterceptor.js:77]: Add comment to this function

File: src/tests/e2e/mocks/cryptoAccounts.js (hunk 4 above)
Comment (timothywangdev) [reply src/tests/e2e/mocks/cryptoAccounts.js:107]: bypass 0xd3ced3b16c8977ed0e345d162d982b899e978588 for allowance setting for metamask, but we do not want to bypass other types of wallets. This further reduced the number of calls.

File: src/tests/e2e/utils/requestInterceptor.js (hunk 1 above)
Comment (timothywangdev) [reply src/tests/e2e/utils/requestInterceptor.js]: fixed

File: src/tests/e2e/utils/requestInterceptor.js (hunk 3 above)
Comment (timothywangdev) [reply src/tests/e2e/utils/requestInterceptor.js]: fixed

File: src/tests/e2e/jest-environment-puppeteer/custom.setup.js (hunk 5 above)
Comment (timothywangdev) [reply src/tests/e2e/jest-environment-puppeteer/custom.setup.js]: fixed

File: src/tests/e2e/directTransferForm.test.e2e.js (hunk 7 above)
Comment (timothywangdev) [reply src/tests/e2e/directTransferForm.test.e2e.js]: fixed

File: src/tests/e2e/utils/requestInterceptor.js (hunk 9 above)
Comment (timothywangdev) [reply src/tests/e2e/utils/requestInterceptor.js]: fixed

**Pull request merged**